### PR TITLE
Optimize startup time, ignore load unConfigured table's column and in…

### LINKED ISSUE #6212: Optimize startup time， control which tables can load when startup
I find the startup time is too slow due to the database full tables metadata scan,  
for our case. the database has 200 tables, we are using micro service, one app loads about 10-20 tables,  it's not acceptable to load all tables on startup

tuohai666:

We have received several similar reports about the slow startup recently, mostly due to loading unconfigured tables for meta data purpose of ShardingSphere.

https://github.com/apache/shardingsphere/issues/6179 : 7000+ tables cost much time.
Similar issues: https://github.com/apache/shardingsphere/issues/6161 , https://github.com/apache/shardingsphere/issues/5755 , https://github.com/apache/shardingsphere/issues/5999 .

Loading table (column and index information) is expensive, there're 7 SQLs interactive for each table.
image

After investigate, I found that for tables unconfigured will not use column or index information any more after loading. So we can omit this process. 

I also checked the relevant issue https://github.com/apache/shardingsphere/issues/4460 , which will load collation for all columns in every table. If a table is unconfigured, the route result must be single route, a single route will not need merge any more. If the resultset does not need to merge, it is irrelevant with cache of collation. So I think we can safely remove the loading of unconfigured tables now.

Note that all unconfigured table names still need to be loaded for routing logic, this can simply obtain from one SQL for each database.

We can accomplish the goal by several steps.

- [ ] Remove the load procedure for unconfigured tables.
- [ ] Remove the refresh procedure for unconfigured tables
- [ ] Remove the multi-thread logic for unconfigured tables. (Configured tables still use multi-thread)
- [ ] Load table names for unconfigured tables.
- [ ] Sync up the new load logic with orchestration module.
- [ ] Update the unit test cases.

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #6212.

Changes proposed in this pull request:
- Remove the load procedure for unconfigured tables. load only table names for unconfigured tables.
- Remove the refresh procedure for unconfigured tables
- Remove the multi-thread logic for unconfigured tables. (Configured tables still use multi-thread)

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java [hunk 1]
```diff
@@ -49,35 +41,33 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 @Slf4j(topic = "ShardingSphere-metadata")
 public final class SchemaMetaDataLoader {
-    
+
```
Comment (tuohai666) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:44]: Don't change original indentation.

Comment (tuohai666) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:44]: image

Comment (xbkaishui) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java]: done

**Comment by xbkaishui** [comment]
Sync up the new load logic with orchestration module. for this task. I think there is no code to change.

**Comment by yu199195** [comment]
@xbkaishui  why remove the multi-thread logic for unconfigured tables？

**Comment by kimmking** [comment]
another question is: when refresh metadata, unconfigured need to be refreshed too.

**Comment by tuohai666** [comment]
Hi @yu199195 , @kimmking , please go through https://github.com/apache/shardingsphere/issues/6212 . unconfigured tables do not need to load the column and index information, we can remove them to improve startup speed.

**Comment by xbkaishui** [comment]
> @xbkaishui why remove the multi-thread logic for unconfigured tables？

for unConfigured tables, only need to load table name,  no need table index and column metadata,  so only one sql will load all tables. no need multi thread logic

**Comment by xbkaishui** [comment]
> another question is: when refresh metadata, unconfigured need to be refreshed too.

yes, for the unConfigured table. now is only load table name

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java [hunk 2]
```diff
@@ -57,27 +49,25 @@
     /**
      * Load schema meta data.
      *
-     * @param dataSource data source
-     * @param maxConnectionCount count of max connections permitted to use for this query
+     * @param dataSource   data source
```
Comment (tuohai666) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:52]: redundant spaces

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java [hunk 3]
```diff
@@ -57,27 +49,25 @@
...
+        return load(dataSource, databaseType, Collections.emptyList());
     }
     
     /**
      * Load schema meta data.
      *
-     * @param dataSource data source
-     * @param maxConnectionCount count of max connections permitted to use for this query
-     * @param databaseType database type
+     * @param dataSource         data source
```
Comment (tuohai666) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:64]: redundant spaces

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java [hunk 4]
```diff
@@ -57,27 +49,25 @@
...
+     * @param databaseType       database type
```
Comment (tuohai666) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:65]: redundant spaces

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataLoader.java [hunk 5]
```diff
@@ -72,6 +68,7 @@
     @SuppressWarnings("unchecked")
     public RuleSchemaMetaData load(final DatabaseType databaseType, final Map<String, DataSource> dataSourceMap, final ConfigurationProperties props,
                                    final ListeningExecutorService executorService) throws SQLException {
```
Comment (tuohai666) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataLoader.java:70]: remove RuleSchemaMetaData if not use anymore

**Pull request closed without merging**